core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -754,7 +743,6 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
-    successfulTaskIdsSet -= tid
```

Comment:
Not handling task failure will cause other issues - we need to keep data structures consistent - cost is secondary to correctness for a performance PR.